mtp_common/notify/email_backend.py

```
@@ -0,0 +1,41 @@
+import logging
+import typing
+
+from django.core.mail.backends.base import BaseEmailBackend
+from django.core.mail.message import EmailMessage
+
+from mtp_common.notify import NotifyClient
+
+logger = logging.getLogger('mtp')
+
+
+class NotifyEmailBackend(BaseEmailBackend):
+    """
+    Email backend that uses NotifyClient to send plain text emails.
+    Should only be used as a fallback because attachments, links and formatting are not supported.
+    The email subject becomes part of the email body text.
+    MTP apps should use the `send_email` task instead with this backend handling emails for core Django functionality.
+    """
+
+    def send_messages(self, email_messages: typing.List[EmailMessage]):
+        for email_message in email_messages:
+            to = email_message.to + email_message.cc + email_message.bcc
```

Don't know if any of the emails we send has BCC field (I had a look and AFACIT we don't use it) so this is probably be an academic comment, sorry 😬 (also this is a fallback email backend that we shouldn't be really using in practice)

I don't think adding the BCC is a good idea, it's a bit sneaky considering that BCC is usually used when you don't want everyone to see these email addresses (for privacy or whatever reason).

In practice it probably doesn't make any difference in our case but maybe we could raise an exception if `email_message.bcc` is not empty saying something like "this email backend doesn't support BCC"?

Just an idea, and as I said, it probably doesn't make any difference at the moment for us.

---

in the end, Notify emails are sent entirely separately to each email address. you can't see who else is getting the email even if they were all originally in the "to" field

---

Oh you're absolutely right because we're looping to send an email notification for each recipient 🤦 - cool, there is no problem then 👌